conda_build/build.py

```diff
@@ -3394,6 +3395,25 @@ def test(
     #     folder destination
     _extract_test_files_from_package(metadata)
 
+    # Remove any previously cached build from the package cache to ensure we
+    # really test the requested build and not some clashing or corrupted build.
+    # (Corruption of the extracted package can happen, e.g., in multi-output
+    # builds if one of the subpackages overwrites files from the other.)
+    # Special case:
+    #   If test is requested for .tar.bz2/.conda file from the pkgs dir itself,
+    #   clean_pkg_cache() will remove it; don't call that function in this case.
+    in_pkg_cache = (
+        not hasattr(recipedir_or_package_or_metadata, "config")
```

Comment:
this is to catch if it's a metadata?

Reply:
Yep, the whole shebang is essentially: "This is a `*.tar.bz`/`*.conda` file (not metadata, or recipe folder) sitting in one of package cache folders".
We could change this to an `isinstance(...)` or something later, but I wanted to retain the previous logic here.